python_ta/transforms/type_inference_visitor.py

```diff
@@ -290,6 +299,32 @@ def visit_ifexp(self, node):
         else:
             node.type_constraints = TypeInfo(Any)
 
+    def visit_comprehension(self, node):
+        if isinstance(node.iter, Name):
```

Comment:
This special case shouldn't be necessary. The `Name` node should have as its type a `TypeVar` that you can look up.